doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,148 @@
+
+.. code-block::
+
+	$ cd ../ && cat t1_reachibility_monitor.dl
+	input relation Links(src: string, dst: string, link_status: bool)
+	output relation ConnectedNodes(str: string, dst: string)
+
+	ConnectedNodes(src, dst) :- Links(src, dst, true).
+	ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+				Links(intermediate_node, dst, true), (src != dst).
+	
+	Links("Menlo Park", "Santa Barbara", true).
+	Links("Menlo Park", "Salt Lake City", true).
+	Links("Santa Barbara", "Los Angeles", true).
+	Links("Los Angeles", "Santa Barbara", true).
+	Links("Los Angeles", "Salt Lake City", true).
+	Links("Salt Lake City", "Los Angeles", true).
+
+	$ ddlog -i t1_reachibility_monitor.dl
+	$ cd t1_reachibility_monitor_ddlog/ && cargo build --release
+	$ ./target/release/t1_reachibility_monitor_cli
+	<clip>
+	>> dump ConnectedNodes;
+	ConnectedNodes{.src = "Los Angeles", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Los Angeles", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Salt Lake City"}
+
+Perfect! Now it looks exactly as what we expected! This is a good time to start doing what this tutorial promised and start writing some C code! We are going to start with something simple yet important. Our initial C program will connect with DDlog program, insert one additional record to :code:`Links` relation, and print the content of :code:`ConnectedNodes` relation. Let's create :code:`reachibility_monitor.c` next to our DDlog program's code. The code is available in the provided :code:`reachibility_monitor_p1.c` file. The further discussion will refer to specific lines in that code.
+
+Let's compile the code first and then delve into discussion of the code and compilation.
+
+.. code-block::
+
+	gcc reachibility_monitor.c t1_reachibility_monitor_ddlog/target/release/libt1_reachibility_monitor_ddlog.a -It1_reachibility_monitor_ddlog/ -lpthread -ldl -lm
+
+
+This concludes the first part of this tutorial. In the next part we are going to modify our program to allow for not only execution of a single command but also being to update existing records using primary key, insert multiple records in a single transaction, and use the `delta` feature of DDlog for a convenient tracking of changes in the output relations.
+
+* add the discussion of the code

```

Comment:
is this supposed to be before the concludes?
You have to explain a lot of stuff about the code: ddog_record, the callback api, the notion of transaction (you could explain transactions even earlier in the cli section).